crates/nu-std/lib/iter.nu

```diff
@@ -119,3 +156,42 @@ export def "iter filter-map" [ # -> list<any>
         $it != null
     }
 }
+
+# Maps a closure to each nested structure and flattens the result
+#
+# # Example
+# ```nu
+# use std ["assert equal" "iter flat-map"]
+#
+# let res = (
+#     [[1 2 3] [2 3 4] [5 6 7]] | iter flat-map {|it| $it | math sum}
+# )
+# ```
+export def "iter flat-map" [ # -> list<any>
+    fn: closure              # the closure to map to the nested structures
+] {
+    each {|it| do $fn $it } | flatten
+}
+
+# Zips two structures and applies a closure to each of the zips
+#
+# # Example
+# ```nu
+# use std ["assert equal" "iter iter zip-with"]
+#
+# let res = (
+#     [1 2 3] | iter zip-with [2 3 4] {|a, b| $a + $b }
+# )
+#
+# assert equal $res [3 5 7]
+# ```
+export def "iter zip-with" [ # -> list<any>
+    other: any               # the structure to zip with
+    fn: closure              # the closure to apply to the zips
+] {
+    $in 
```

Comment:
this `$in` can be removed :+1: 
you can leave it if you prefer being extra explicit but then why not add it too in `iter flat-map`? :open_mouth:

Reply:
let me remove it for consistency